sbt-java-gen/src/main/scala/Fs2GrpcServicePrinter.scala

```diff
@@ -73,28 +78,74 @@ class Fs2GrpcServicePrinter(service: ServiceDescriptor, di: DescriptorImplicits)
       .outdent
 
   private[this] def serviceTrait: PrinterEndo =
-    _.add(s"trait ${serviceName}Fs2Grpc[F[_]] {").indent.call(serviceMethods).outdent.add("}")
+    _.add(s"trait $serviceNameFs2[F[_], $Ctx] {").
+      indent.
+      call(serviceMethods).
+      outdent.
+      add("}")
 
   private[this] def serviceObject: PrinterEndo =
-    _.add(s"object ${serviceName}Fs2Grpc {").indent.call(serviceClient).call(serviceBinding).outdent.add("}")
+    _.add(s"object $serviceNameFs2 {").
+      indent.
+      newline.
+      call(serviceTraitMeta).
+      newline.
+      call(serviceClientMeta).
+      newline.
+      call(serviceBindingMeta).
+      newline.
+      call(serviceClient).
+      newline.
+      call(serviceBinding).
+      outdent.
+      add("}")
 
   private[this] def serviceClient: PrinterEndo = {
     _.add(
-      s"def stub[F[_]: $ConcurrentEffect](channel: $Channel, callOptions: $CallOptions = $CallOptions.DEFAULT): ${serviceName}Fs2Grpc[F] = new ${serviceName}Fs2Grpc[F] {").indent
+      s"def client[F[_]: $ConcurrentEffect, $Ctx](channel: $Channel, f: $Ctx => $Metadata, callOptions: $CallOptions = $CallOptions.DEFAULT): $serviceNameFs2[F, $Ctx] = new $serviceNameFs2[F, $Ctx] {")
+      .indent
       .call(serviceMethodImplementations)
       .outdent
       .add("}")
   }
 
   private[this] def serviceBinding: PrinterEndo = {
     _.add(
-      s"def bindService[F[_]: $ConcurrentEffect](serviceImpl: ${service.name}Fs2Grpc[F]): $ServerServiceDefinition = {").indent
+      s"def service[F[_]: $ConcurrentEffect, $Ctx](serviceImpl: $serviceNameFs2[F, $Ctx], f: $Metadata ⇒ $Ctx): $ServerServiceDefinition = {")
```

Comment:
Unicode arrows are deprecated in Scala 2.13.  I didn't scan the review for others:

```suggestion
      s"def service[F[_]: $ConcurrentEffect, $Ctx](serviceImpl: $serviceNameFs2[F, $Ctx], f: $Metadata => $Ctx): $ServerServiceDefinition = {")
```